src/thin_arc.rs

```diff
@@ -29,7 +28,7 @@ use crate::header::HeaderSliceWithLengthUnchecked;
 /// via `HeaderSliceWithLengthProtected`.
 #[repr(transparent)]
 pub struct ThinArc<H, T> {
-    ptr: ptr::NonNull<ArcInner<HeaderSliceWithLengthProtected<H, [T; 0]>>>,
+    ptr: ptr::NonNull<ArcInner<HeaderSlice<HeaderWithLength<H>, [T; 0]>>>,
```

Comment:
ah, I see, yeah that will have the same layout.

drop a comment here about the layouts being the same wrt CoerceUnsized?

Reply:
I'm not sure where exactly `CoerceUnsized` would still be relevant. The wrapping types are all `repr(C)`/`repr(transparent)` anyway, and the layout of `[T; N]` and `[T]` is well-defined (in terms of size&align of `T`), too.

I did add some comments though, feel free to give feedback.

Reply:
Oh, you're right, there's no CoerceUnsized here, it's just a regular layout cast. But yeah those comments are still appreciated!!